packages/kit/src/runtime/server/endpoint.js

```diff
@@ -11,7 +11,7 @@ import { method_not_allowed } from './utils.js';
 export async function render_endpoint(event, mod, state) {
 	const method = /** @type {import('types').HttpMethod} */ (event.request.method);
 
-	let handler = mod[method];
+	let handler = mod[method] || mod.all;
 
 	if (!handler && method === 'HEAD') {
 		handler = mod.GET;
```

Comment:
This might catch people off guard, if someone has a HEAD handler and then they add an all handler, it'll implicitly switch to using the all handler.

Should we try to keep the `HEAD` uses `GET` without the body semantics or just document that effect of the all handler?

Reply:
I think it would be good to document the effect of the all handler.
At the moment, the `HEAD` handler is automatic and can't be exported by a user (although there's a bug right now where it is being exported during build). https://github.com/sveltejs/kit/pull/9753 fixes that.